src/bd_rhapsody/bd_rhapsody_sequence_analysis/script.py

```diff
@@ -0,0 +1,235 @@
+                par[argument["clean_name"]] = [ os.path.abspath(f) for f in par[argument["clean_name"]] ]
+            else:
+                par[argument["clean_name"]] = os.path.abspath(par[argument["clean_name"]])
+    
+    return par
+
+def generate_config(par: dict[str, Any], config) -> str:
+    content_list = [strip_margin(f"""\
+        |#!/usr/bin/env cwl-runner
+        |
+        |cwl:tool: rhapsody
+        |""")]
+
+    for argument in config["arguments"]:
+        arg_info = argument.get("info") or {}
+        config_key = arg_info.get("config_key")
+        if par[argument["clean_name"]] and config_key:
+
+            if argument["type"] == "file":
+                str = strip_margin(f"""\
+                    |{config_key}:
+                    |""")
+                if isinstance(par[argument["clean_name"]], list):
+                    for file in par[argument["clean_name"]]:
+                        str += strip_margin(f"""\
+                            | - class: File
+                            |   location: "{file}"
+                            |""")
+                else:
+                    str += strip_margin(f"""\
+                        |   class: File
+                        |   location: "{par[argument["clean_name"]]}"
+                        |""")
+                content_list.append(str)
+            else:
+                content_list.append(strip_margin(f"""\
+                    |{config_key}: {par[argument["clean_name"]]}
+                    |"""))
+
+    ## Write config to file
+    return ''.join(content_list)
+
+def generate_config_file(par: dict[str, Any], config: dict[str, Any], temp_dir: str) -> str:
+    config_file = os.path.join(temp_dir, "config.yml")
+    config_content = generate_config(par, config)
+    with open(config_file, "w") as f:
+        f.write(config_content)
+    return config_file
+
+def generate_cwl_file(meta: dict[str, Any], dir: str) -> str:
+    # create cwl file (if need be)
+    orig_cwl_file=os.path.join(meta["resources_dir"], "rhapsody_pipeline_2.2.1_nodocker.cwl")
+
+    # Inject computational requirements into pipeline
+    if meta["memory_mb"] or meta["cpus"]:
+        cwl_file = os.path.join(dir, "pipeline.cwl")
+
+        # Read in the file
+        with open(orig_cwl_file, 'r') as file :
+            cwl_data = file.read()
+
+        # Inject computational requirements into pipeline
+        if meta["memory_mb"]:
+            memory = int(meta["memory_mb"]) - 2000 # keep 2gb for OS
+            cwl_data = re.sub('"ramMin": [^\n]*[^,](,?)\n', f'"ramMin": {memory}\\1\n', cwl_data)
+        if meta["cpus"]:
+            cwl_data = re.sub('"coresMin": [^\n]*[^,](,?)\n', f'"coresMin": {meta["cpus"]}\\1\n', cwl_data)
+
+        # Write the file out again
+        with open(cwl_file, 'w') as file:
+            file.write(cwl_data)
+    else:
+        cwl_file = orig_cwl_file
+
+    return os.path.abspath(cwl_file)
```

Comment:
I prefer to return early 🤷 
```suggestion
    orig_cwl_file=os.path.join(meta["resources_dir"], "rhapsody_pipeline_2.2.1_nodocker.cwl")
    if not meta["memory_mb"] and not meta["cpus"]:
        return os.path.abspath(orig_cwl_file)
     
    # Inject computational requirements into pipeline
    cwl_file = os.path.join(dir, "pipeline.cwl")

    # Read in the file
    with open(orig_cwl_file, 'r') as file :
        cwl_data = file.read()

    # Inject computational requirements into pipeline
    if meta["memory_mb"]:
        memory = int(meta["memory_mb"]) - 2000 # keep 2gb for OS
        cwl_data = re.sub('"ramMin": [^\n]*[^,](,?)\n', f'"ramMin": {memory}\\1\n', cwl_data)
    if meta["cpus"]:
        cwl_data = re.sub('"coresMin": [^\n]*[^,](,?)\n', f'"coresMin": {meta["cpus"]}\\1\n', cwl_data)

    # Write the file out again
    with open(cwl_file, 'w') as file:
        file.write(cwl_data)
        
    return os.path.abspath(cwl_file)
```